src/main/scala/com/mozilla/telemetry/streaming/EventPingEvents.scala

```diff
@@ -0,0 +1,149 @@
+    verify()
+  }
+
+  def main(args: Array[String]): Unit = {
+    val opts = new Opts(args)
+
+    val spark = SparkSession.builder()
+      .appName("Event Ping Events")
+      .config("spark.streaming.stopGracefullyOnShutdown", "true")
+      .getOrCreate()
+
+    opts.kafkaBroker.get match {
+      case Some(_) => writeStreamingEvents(spark, opts)
+      case None => writeBatchEvents(spark, opts)
+    }
+  }
+
+  def writeStreamingEvents(spark: SparkSession, opts: EventPingEvents.Opts): Unit = {
+    val pings = spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", opts.kafkaBroker())
+      .option("failOnDataLoss", opts.failOnDataLoss())
+      .option("kafka.max.partition.fetch.bytes", 8 * 1024 * 1024) // 8MB
+      .option("spark.streaming.kafka.consumer.cache.maxCapacity", kafkaCacheMaxCapacity)
+      .option("subscribe", TelemetryKafkaTopic)
+      .option("startingOffsets", opts.startingOffsets())
+      .load()
+
+    val outputPath = opts.outputPath()
+
+    explodeEvents(pings.select("value"))
+      .repartition(1)
+      .writeStream
+      .queryName(QueryName)
+      .format("parquet")
+      .option("path", s"${outputPath}/${outputPrefix}")
+      .option("checkpointLocation", opts.checkpointPath())
+      .partitionBy("submission_date_s3", "docType")
+      .start()
+      .awaitTermination()
+
+  }
+
+  def writeBatchEvents(spark: SparkSession, opts: EventPingEvents.Opts): Unit = {
+    implicit val sc = spark.sparkContext
+    import spark.implicits._
+
+    datesBetween(opts.from(), opts.to.get).foreach { currentDate =>
+      val pings = MozDataset("telemetry")
+        .where("sourceName") { case "telemetry" => true }
+        .where("docType") { case docType if allowedDocTypes.contains(docType) => true }
+        .where("appName") { case appName if allowedAppNames.contains(appName) => true }
+        .where("submissionDate") { case date if date == currentDate => true }
+        .records(opts.fileLimit.get)
+        .map(_.toByteArray)
+
+      val pingsDataframe = pings.toDF("value")
+
+      val outputPath = s"${opts.outputPath()}/${outputPrefix}/submission_date_s3=$currentDate/docType=event"
+
+      logger.info(s"Processing and event pings for $currentDate")
+
+      explodeEvents(pingsDataframe)
+        .write
+        .option("maxRecordsPerFile", opts.maxRecordsPerFile())
+        .mode("overwrite")
+        .parquet(outputPath)
+
+      logger.info(s"Event ping events written for $currentDate")
+    }
+
+    spark.stop()
+  }
+
+  private[streaming] def explodeEvents(messages: DataFrame): Dataset[EventRow] = {
+    import messages.sparkSession.implicits._
+
+    messages.flatMap(v => {
+      //try {
+        val m = Message.parseFrom(v.get(0).asInstanceOf[Array[Byte]])
+        val fields = m.fieldsAsMap
+        val docType = fields.getOrElse("docType", "").asInstanceOf[String]
+        if (!allowedDocTypes.contains(docType)) {
```

Comment:
I'd prefer to reverse the order of the branches here so we can remove the `!` and avoid a double negative.